build.sbt

```diff
@@ -309,19 +313,50 @@ lazy val core = (project in file("core"))
 lazy val root = RootProject(file("."))
 
 lazy val chisel = (project in file("."))
-  .enablePlugins(ScalaUnidocPlugin)
   .settings(commonSettings: _*)
   .settings(chiselSettings: _*)
-  .settings(publishSettings: _*)
+  .settings(
+    // Published as part of unipublish
+    publish / skip := true
+  )
   .settings(usePluginSettings: _*)
   .dependsOn(macros)
   .dependsOn(core)
   .dependsOn(firrtl)
   .aggregate(macros, core, plugin, firrtl)
   .settings(warningSuppression: _*)
   .settings(fatalWarningsSettings: _*)
+
+
+def addUnipublishDeps(proj: Project)(deps: Project*): Project =
+  deps.foldLeft(proj) { case (p, dep) =>
+    //p.dependsOn(dep % "compile-internal")
```

Comment:
```suggestion
```
Surprisingly, we don't need an actual `.dependsOn`, the `mappings` stuff makes everything work as expected.